src/e2ir.d

```diff
@@ -5701,6 +5704,24 @@ Symbol *toStringSymbol(const(char)* str, size_t len, size_t sz, bool comdat = fa
             OutBuffer buf;
             buf.writestring("__");
             mangleToBuffer(se, &buf);   // recycle how strings are mangled for templates
+
+            if (buf.offset >= 32 + 2)
+            {   // Replace long string with hash of that string
```

Comment:
cc @wilzbach - we should tolerate this in the compiler for now, but not in druntime or phobos